db/internal_stats.cc

```diff
@@ -755,50 +889,78 @@ void InternalStats::DumpCFStats(std::string* value) {
     total_files += files;
     total_files_being_compacted += files_being_compacted[level];
     if (comp_stats_[level].micros > 0 || files > 0) {
-      stats_sum.Add(comp_stats_[level]);
+      compaction_stats_sum->Add(comp_stats_[level]);
       total_file_size += vstorage->NumLevelBytes(level);
       double w_amp =
-          (comp_stats_[level].bytes_read_non_output_levels == 0) ? 0.0
-          : static_cast<double>(comp_stats_[level].bytes_written) /
-            comp_stats_[level].bytes_read_non_output_levels;
-      PrintLevelStats(buf, sizeof(buf), "L" + ToString(level), files,
-                      files_being_compacted[level],
-                      static_cast<double>(vstorage->NumLevelBytes(level)),
-                      compaction_score[level],
-                      w_amp, comp_stats_[level]);
-      value->append(buf);
+          (comp_stats_[level].bytes_read_non_output_levels == 0)
+              ? 0.0
+              : static_cast<double>(comp_stats_[level].bytes_written) /
+                    comp_stats_[level].bytes_read_non_output_levels;
+      auto level_stats = std::map<LevelStatType, double>();
+      PrepareLevelStats(&level_stats, files, files_being_compacted[level],
+                        static_cast<double>(vstorage->NumLevelBytes(level)),
+                        compaction_score[level], w_amp, comp_stats_[level]);
+      (*levels_stats)[level] = level_stats;
     }
   }
   uint64_t flush_ingest = cf_stats_value_[BYTES_FLUSHED];
   uint64_t add_file_ingest = cf_stats_value_[BYTES_INGESTED_ADD_FILE];
   uint64_t curr_ingest = flush_ingest + add_file_ingest;
+  // Cumulative summary
+  double w_amp = compaction_stats_sum->bytes_written /
+                 static_cast<double>(curr_ingest + 1);
+  // Stats summary across levels
+  auto level_stats = std::map<LevelStatType, double>();
```

Comment:
Also what do you think about renaming this to `sum_stats` since it's not actually a level stats